programs/ssl/ssl_server2.c

```diff
@@ -2020,7 +2020,6 @@ int main( int argc, char *argv[] )
     if( received_sigterm )
     {
         mbedtls_printf( " interrupted by SIGTERM\n" );
-        ret = 0;
```

Comment:
With this change to never clear `ret`, for a normal test run (no failures, no timing weirdness), the server logs now contain
```
interrupted by SIGTERM
Last error was: -0x45 - NET - The context is invalid, eg because it was free()ed
```
Although the message mentions an error, this error was intended: the connection is supposed to have been closed mid-connection. I think it would be best to keep suppressing the `MBEDTLS_ERR_NET_INVALID_CONTEXT` here.

Reply:
Under what circumstances do you get this message? It don't this this happens when running `ssl-opt.sh` because otherwise `Default` would fail. If it's when running the server and killing in manually, then I would tend not to care: I assume a human can make sense of this error, so I chose to care only about stupid scripts - or more precisely, only those stupid scripts that are part of our test system.

Reply:
Also, a motivation for not clearing `ret` except after `net_accept()` is that, since I don't expect `SIGTERM` to be received at any other place than `net_accept()` under normal (successful) testing circumstances, I'm afraid reseting it in other places could hide errors. Perhaps it doesn't, but I couldn't come up with a strong argument to that effect.

As much as I hate false positives on the CI, the last thing I hate more is false negatives. So while I'm not sure reseting `ret` here could hide errors, I don't want to take any unnecessary risk.

Reply:
Run `ssl-opt.sh -p` and check where you get `interrupted by SIGTERM` vs `interrupted by signal`. Under `Default`, the signal is caught at the `interrupted by signal` location.

Reply:
@mpg I'm not sure we're understanding each other here. [This](https://github.com/gilles-peskine-arm/mbedtls/commit/c799e9ca4ff43ed86fbf1ce62350bc584b42d664) is what looks right to me. No clearing of errors in general, but do clear an the error resulting from interrupting the network operation, and use the same treatment in both places where a signal is detected.

Reply:
@gilles: thanks for letting me know about `ssl-opt.sh -p`, I wasn't aware of that option, and it help fix my misconception: signals can indeed be received outside of `net_accept()` during normal (successful) testing.

Also, regarding my fear that reseting `ret` when the signal is received outside `accept()`, it looks like currently we don't have tests that ensure that the server does not see the client disconnecting unexpectedly, and if in the future we want to add some, we can still do so by checking for the "interrupted" message rather than `ret`.

So I'll adopt your fix and take the occasion to enhance the messages.